### ISSUE TITLE
netflow codec doesn't seem to work with logstash 2.3.4

### ISSUE THREAD

**Issue by zerosignal0** [logstash-plugins/logstash-codec-netflow#41]
Hello,

I have opened a thread on the elastic.co forums on this issues, however was able to validate with my configuration that it seems 2.3.4 logstash is not able to be loaded using the latest netflow codec.  I had validated with a few known working 2.1.0 netflow configuration files procured from the IRC channel, however these would not load as well in logstash 2.3.4.  Upon downgrading to 2.2.4 I can successfully use the netflow codec without issue.

https://discuss.elastic.co/t/question-regarding-new-logstash-netflow-config

**Comment by jorritfolmer** [comment]
The config you posted on the forum should be expected to work. However, this doesn't seem to be the config file referenced in the error message, `20-logstash-staticfile-netflow.conf` Can you post that one too?

I'm also unable to reproduce this on a (Red Hat 7) vm with Logstash 2.3.4-1 and the config you posted.
```
# /opt/logstash/bin/logstash --configtest /etc/logstash/conf.d/logstash-netflow.conf 
Configuration OK
#
```

**Comment by renevdm** [comment]
I have the same issue. Opened https://github.com/elastic/logstash/issues/5608 for this, but closed it as it seems to be a duplicate of this one.

**Comment by marke72** [comment]
I have the same issue as well. I updated from 2.3.3 to 2.3.4 and got this exact error when starting logstash. I tried with a fresh install of logstash but still the same error. My codec version is 2.1.0 which is what comes with the current version of logstash

**Comment by jorritfolmer** [comment]
Ah I'm getting the same error now when starting logstash instead of only `--configtest`

**Comment by jorritfolmer** [comment]
It appears that [this commit to the BinData library in march 26nd](https://github.com/dmendel/bindata/commit/0e0e636364647e14f03853993490d10a5edf30f5) expands the number of reserved words, of which `fields` is one. Since Logstash 2.3.4 now includes this particular version of BinData, we get this error.

**Comment by jorritfolmer** [comment]
Ok fixed it with commit 5eae8d5f57ce25977639992955da6c247689e632.
And published a new logstash-codec-netflow-2.1.1 gem on rubygems 

You can update the netflow codec to v2.1.1 through:

```
/opt/logstash/bin/logstash-plugin uninstall logstash-codec-netflow
```

and then:

```
/opt/logstash/bin/logstash-plugin install --version 2.1.1 logstash-codec-netflow
```

**Comment by marke72** [comment]
I just installed the update and it seems to be working great! Thanks for the quick fix!

**Comment by renevdm** [comment]
Great! Works like a charm again! :-)
Thanks for making an update available this quick!

**Comment by suyograo** [comment]
Fixed in 2.1.1

**Comment by Oydin** [comment]
_Good day! Maybe some of us faced with this problem. I use:_
**- Red Hat 7
- logstash-all-plugins-2.3.4-1.noarch.rpm
- java-1.8.0-openjdk.x86_64 1:1.8.0.91-1.b14.el7_2
- elasticsearch-2.3.4.rpm
- kibana-4.5.3-1.x86_64.rpm**
_The web site is available, full green. No extra configuration is did. I used the two commands:_
1. #/opt/logstash/bin/plugin uninstall logstash-codec-netflow
2. #/opt/logstash/bin/plugin install --version 2.1.1 logstash-codec-netflow
_Then i configured conf.d/t.conf:_
input {
     udp {
       port => 9996
       type => "netflow"
       codec => netflow {
         versions => [5,9,10]
       }
     }
}

output {
 if [type] == "netflow" {
  elasticsearch {
     hosts => localhost
     index => "netflow-%{+YYYY.MM.dd}"
     }
   }
}
_After 20 minutes of logstash, logstash.log increased to 20 megabytes and consisted only of the records of this type:_
:message=>"No matching template for flow id 256", :level=>:warn}
:message=>"No matching template for flow id 265", :level=>:warn}
:message=>"No matching template for flow id 263", :level=>:warn}
:message=>"No matching template for flow id 260", :level=>:warn}
_I also noticed that if you leave Logstash to work in this state, after 6 -7 hours Kibana stops working and does not start. At the same time remove logstash and Kobanov fails. Throws error_ "No Match for argument: logstash No Packages marked for removal".  
logstash.log:
{:timestamp=>"2016-07-29T09:16:32.058000+1000", :message=>#<LogStash::PipelineReporter::Snapshot:0x13dde831 @data={:events_filtered=>0, :events_consumed=>0, :worker_count=>2, :inflight_count=>0, :worker_states=>[{:status=>"dead", :alive=>false, :index=>0, :inflight_count=>0}, {:status=>"dead", :alive=>false, :index=>1, :inflight_count=>0}], :output_info=>[{:type=>"elasticsearch", :config=>{"hosts"=>"localhost", "index"=>"netflow-%{+YYYY.MM.dd}", "__ALLOW_ENV__"=>false}, :is_multi_worker=>false, :events_received=>0, :workers=><Java::JavaUtilConcurrent::CopyOnWriteArrayList:-1580330739 [<LogStash::Outputs::ElasticSearch hosts=>["localhost"], index=>"netflow-%{+YYYY.MM.dd}", codec=><LogStash::Codecs::Plain charset=>"UTF-8">, workers=>1, manage_template=>true, template_name=>"logstash", template_overwrite=>false, flush_size=>500, idle_flush_time=>1, doc_as_upsert=>false, max_retries=>3, script_type=>"inline", script_var_name=>"event", scripted_upsert=>false, retry_max_interval=>2, retry_max_items=>500, retry_on_conflict=>1, action=>"index", ssl_certificate_verification=>true, sniffing=>false, sniffing_delay=>5, discovery=>"zen", aws_protocol=>"https">]>, :busy_workers=>1}], :thread_info=>[], :stalling_threads_info=>[]}>, :level=>:warn}
_It has to be used each time snapshot..._
**I do not understand what I'm doing wrong. If someone had the same problem, I hope together we can find the truth :)**
